osu.Game/Screens/Play/Player.cs

```diff
@@ -627,6 +630,10 @@ public override void OnEntering(IScreen last)
 
             foreach (var mod in Mods.Value.OfType<IApplicableToHUD>())
                 mod.ApplyToHUD(HUDOverlay);
+
+            Beatmap.Value.Track.ResetSpeedAdjustments();
+            foreach (var mod in Mods.Value.OfType<IApplicableToTrack>())
+                mod.ApplyToTrack(Beatmap.Value.Track);
```

Comment:
In the failing tests, doing this causes the adjustment to get re-applied twice because of this:

https://github.com/ppy/osu/blob/7e1844ed773368a4b932ea0e2d7fc87fa0fc53b4/osu.Game/Tests/Visual/PlayerTestScene.cs#L33

Disabling this back again locally makes them pass, but this makes me quite uneasy... Maybe that `true` override is no longer necessary with the recent changes?

Reply:
i would hope it's not needed.. but also hope that it didn't break anything.